Add support for multiple key shares (#26445)
openssl/openssl: CHANGES.md

### Changes between 3.4 and 3.5 [xx XXX xxxx]

> * For TLSv1.3: Add capability for a client to send multiple key shares. Extend reach of

Reviewer: ```suggestion
* For TLSv1.3: Add capability for a client to send multiple key shares. Extend the scope of
```

>   `SSL_OP_CIPHER_SERVER_PREFERENCE` to also include server-side key exchange group selection.

Reviewer: ```suggestion
  `SSL_OP_CIPHER_SERVER_PREFERENCE` to cover server-side key exchange group selection.
```

>   Extend server-side key exchange group selection algorithm and related group list syntax

Reviewer: ```suggestion
  Extend the server-side key exchange group selection algorithm and related group list syntax
```

>   to allow levels of group priorities e.g. to prioritize (hybrid-)KEMs.

Reviewer: ```suggestion
  to support multiple group priorities, e.g. to prioritize (hybrid-)KEMs.
```
